src/courses/mappers/04.md

```diff
@@ -0,0 +1,113 @@
+---
+order: 4
+next: 05.md
+title: Understanding InspecJS & OHDF Converters
+author: Charles Hu
+---
+
+## What is InspecJS?
+
+InspecJS is a library that provides schema definitions, classes, and utilities for OHDF file handling. InspecJS plays a pivotal role in the contextualization process where it converts the individual statuses for each finding into an overall status for the control.
+
+## What is OHDF Converters?
+
+OHDF Converters is a custom data normalization library which leverages OHDF mappers for transforming various security data formats to and from OHDF. It is currently integrated into [Heimdall2](https://github.com/mitre/heimdall2) and the [SAF CLI](https://github.com/mitre/saf), which collectively are part of the [Security Automation Framework (SAF)](https://saf.mitre.org/#/), a set of tools and processes which aim to standardize and ease security compliance testing and verification within an automated build pipeline.
+
+### Features
+
+- Leverages OHDF mappers to transform security data formats to and from OHDF.
+- Special keyword functions provided by the <i>base-converter</i> class.
+- Fingerprinting services for automatic data format identification.
+
+### Keywords
+
+<i>base-converter</i> is an underlying class which enables \*-to-HDF mapping in OHDF Converters. It defines \*-to-HDF mappers and provides critical tools in the form of keywords which allow for the construction of such mappers. All \*-to-HDF mappers inherit from this class and therefore have access to the keywords that this class provides. You should aim to utilize these keywords to their fullest potential to ease and simplify mapper development. The provided tools are as follows:
+
+- `path`: Define JSON object path to go to. Paths are found recursively.
+  - Use:
+  ```
+  path: PATH_AS_STRING
+  ```
+  - Example:
+  ```
+  // Attribute 'id' will be set as whatever JSON object attribute 'vulnerability.id' is
+  id: {path: 'vulnerability.id'}
+  ```
+
+- `transformer`: Execute given code sequence; operates similar to an anonymous function.
+  - Use:
+  ```
+  transformer: (PARAMETER: TYPE): OUTPUT_TYPE => {CODE_TO_EXECUTE}
+  ```
+  - Example:
+  ```
+  // Attribute 'code' will be set as the returned stringified JSON object of input 'vulnerability'
+  code: {
+    transformer: (vulnerability: Record<string, unknown>): string => {
+      return JSON.stringify(vulnerability, null, 2);
+    }
+  }
```

Comment:
Should this include a `path: 'vulnerability`? As a reader, it would look like the parameter name is specifying where this comes from.